developer-docs/user-proxy.md

```diff
@@ -0,0 +1,118 @@
+# User Proxy authentication
+
+## Problem description
+
+In Provena, services are deployed in a microservice type architecture. Sometimes, there are privileged operations which only other specific services should be able to invoke in a given service.
+
+---
+
+**Key problem**: when service A makes a service-only request to service B, it both needs to 1. use a JWT Bearer token which proves it _is_ service A and 2. pass through the user's roles and other information so that service B can enforce authorisation and other constraints correctly.
+
+---
+
+A pertinent example is creating a dataset. Currently, the `data-store-api` is responsible for this task. Consider the high-level steps needed to undertake this process
+
+1. validate the item doesn't exist (requires fetching from the registry, through the `registry-api`)
```

Comment:
for creating a dataset what are we validating that doesn't exist if we are creating something here?

Reply:
true - woops